src/apps/answers/crud/answers.py

```diff
@@ -234,10 +234,21 @@ async def get_respondents_submit_dates(
         query = query.where(func.date(AnswerSchema.created_at) >= filters.from_date)
         query = query.where(func.date(AnswerSchema.created_at) <= filters.to_date)
         query = query.where(AnswerSchema.applet_id == applet_id)
+
         if filters.respondent_id:
             query = query.where(AnswerSchema.respondent_id == filters.respondent_id)
         if filters.target_subject_id:
             query = query.where(AnswerSchema.target_subject_id == filters.target_subject_id)
+
+        if filters.activity_or_flow_id:
+            activity_or_flow_id = str(filters.activity_or_flow_id)
+
+            query = query.where(
+                or_(
+                    AnswerSchema.id_from_history_id(AnswerSchema.activity_history_id) == activity_or_flow_id,
+                    AnswerSchema.id_from_history_id(AnswerSchema.flow_history_id) == activity_or_flow_id,
+                )
+            )
```

Comment:
When filtering answer results by activity ID, we should exclude activities that were completed as part of a flow. This is a common pattern we use elsewhere in the BE code:
```suggestion
            query = query.where(
                case(
                    (
                        AnswerSchema.flow_history_id.isnot(None),
                        AnswerSchema.id_from_history_id(AnswerSchema.flow_history_id) == activity_or_flow_id,
                    ),
                    else_=AnswerSchema.id_from_history_id(AnswerSchema.activity_history_id) == activity_or_flow_id,
                )
            )
```

Reply:
Alright! Adding that change